rebasehelper/tests/public_api/test_specfile.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you rebase your package to the latest version
+# Copyright (C) 2013-2019 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hráček <phracek@redhat.com>
+#          Tomáš Hozza <thozza@redhat.com>
+#          Nikola Forró <nforro@redhat.com>
+#          František Nečas <fifinecas@seznam.cz>
+
+import os
+import shutil
+
+import pytest  # type: ignore
+
+from rebasehelper.specfile import SpecFile
+from rebasehelper.tests.conftest import SPEC_FILE, TEST_FILES_DIR
+
+
+@pytest.mark.public_api
+class TestSpecFile:
+    def test_constructor(self, workdir):
+        shutil.copy(os.path.join(TEST_FILES_DIR, SPEC_FILE), workdir)
+        # check that no mandatory argument was added
+        SpecFile(SPEC_FILE)
+        # check that arguments were not renamed or removed
+        SpecFile(path=SPEC_FILE, sources_location=workdir, predefined_macros=None)
+
+    def test_update_changelog(self, spec_object):
+        entry = 'test message'
+        spec_object.update_changelog(changelog_entry=entry)
+        assert entry in spec_object.spec_content.section('%changelog')[1]
+        spec_object.update_changelog(entry)
+        assert entry in spec_object.spec_content.section('%changelog')[1]
+
+    def test_set_version(self, spec_object):
+        NEW_VERSION = '1.2.3.4.5'
+        spec_object.set_version(version=NEW_VERSION)
+        assert spec_object.header.version == NEW_VERSION
+
+        NEW_VERSION = '1.2.3.4'
+        spec_object.set_version(NEW_VERSION)
+        assert spec_object.header.version == NEW_VERSION
+
+    def test_get_version(self, spec_object):
+        assert spec_object.get_version() == '1.0.2'
+
+    def test_set_release_number(self, spec_object):
+        spec_object.set_release_number(release=1)
+        assert '1' == spec_object.get_release()
+
+        spec_object.set_release_number(2)
+        assert '2' == spec_object.get_release()
+
+    def test_set_release(self, spec_object):
+        spec_object.set_release(release=1)
+        assert '1' == spec_object.get_release()
+
+        spec_object.set_release(2)
+        assert '2' == spec_object.get_release()
+
+    def test_get_sources(self, spec_object):
+        assert isinstance(spec_object.get_sources(), list)
```

Comment:
Again, an inconsistency. You check exact value everywhere else, but only type here. It should be either this or that. I think checking type should be enough, it will make the tests less dependent on potential changes in the testing spec file (but perhaps it would be best to make a separate spec file for API tests).

Reply:
Hm, the thing is that with some methods (`update_changelog`) you can't really only check the type. The same goes for `set_release_number` and `set_version`. With these methods, we probably want t ocheck that the release was really set. I could check the exact value here as well but that already gets taken care of by our unit tests so it would be pointless. There are a lot of sources so listing them all out is inconvenient (this could be solved by having a separate testing spec)

Reply:
> you can't really only check the type

Yes, you can. These are API (interface) tests, they are supposed to complement unit/functional tests that test the actual functionality.

Reply:
Okay, so lets just strictly check the return value type and the arguments in this test suite, that will probably be better

Reply:
While not strictly necessary, it would be more consistent to check return type of every method, even those that return `None` or instance of `SpecFile` (the constructor).